sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/WhitespaceBeforeArrayInitializerCheck.java

```diff
@@ -0,0 +1,78 @@
+package com.github.sevntu.checkstyle.checks.coding;
+
+import com.puppycrawl.tools.checkstyle.api.Check;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * <p>
+ * This checks enforces whitespace before array initializer
+ * </p>
+ * Examples:
+ * This code is perfectly valid:
+ * <p>
+ * <code>
+ * <pre>
+ * int[] ints = new int[] {
+ *           0, 1, 2, 3
+ * };
+ * </pre>
+ * </code>
+ * </p>
+ * But this violates check:
+ * <p>
+ * <code>
+ * <pre>
+ * int[] ints = new int[]{0, 1, 2, 3};
+ * </pre>
+ * </code>
+ * </p>
+ *
+ * @author <a href="mailto:piotr.listkiewicz@gmail.com">liscju</a>
+ */
+public class WhitespaceBeforeArrayInitializerCheck extends Check {
+
+    /**
+     * Violation message key.
+     */
+    public static final String MSG_KEY = "whitespace.before.array.initializer";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {TokenTypes.ARRAY_INIT};
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (ast.getPreviousSibling() != null && ast.getPreviousSibling().getType() == TokenTypes.ARRAY_DECLARATOR) {
+            DetailAST arrayDeclaratorAST = ast.getPreviousSibling();
+            checkDistanceBetweenTokens(ast, getEndOfArrayDeclarator(arrayDeclaratorAST));
+        } else {
+            //assert ast.getParent().getType() == TokenTypes.ASSIGN;
```

Comment:
@mkordas , agree with you but do not know how to make algorithm that distinguish commented code and just a comment with some code in it.